src/it/java/teammates/it/storage/sqlapi/CoursesDbIT.java

```diff
@@ -124,4 +124,207 @@ public void testGetTeamsForCourse() throws InvalidParametersException, EntityAlr
+
+        // Attempt to use SQL commands in courseId field
+        String courseId = "test' OR 1 = 1; --";
+        Course actual = coursesDb.getCourse(courseId);
+        assertEquals(null, actual);
+    }
+
+    @Test
+    public void testSqlInjectionInUpdateCourse() throws Exception {
+        ______TS("SQL Injection test in updateCourse");
+
+        Course course = new Course("course-id", "name", "UTC", "institute");
+        coursesDb.createCourse(course);
+
+        // The system should treat the input as a plain text string
+        String newName = "newName'; DROP TABLE courses; --";
+        course.setName(newName);
+        coursesDb.updateCourse(course);
+        Course actual = coursesDb.getCourse("course-id");
+        assertEquals(newName, actual.getName());
+    }
+
+    @Test
+    public void testSqlInjectionInDeleteCourse() throws Exception {
+        ______TS("SQL Injection test in deleteCourse");
+
+        Course course = new Course("course-id", "name", "UTC", "institute");
+        coursesDb.createCourse(course);
+
+        String name = "newName'; DELETE FROM courses; --";
+        Course injectionCourse = new Course("course-id-injection", name, "UTC", "institute");
+        coursesDb.createCourse(injectionCourse);
+
+        coursesDb.deleteCourse(injectionCourse);
+        Course actualInjectionCourse = coursesDb.getCourse("course-id-injection");
+
+        // The course should be deleted
+        assertEquals(null, actualInjectionCourse);
+
+        // All other courses should not be deleted
+        Course actualCourse = coursesDb.getCourse("course-id");
+        assertEquals(course, actualCourse);
+    }
+
+    @Test
+    public void testSqlInjectionInCreateSection() throws Exception {
+        ______TS("SQL Injection test in createSection");
+
+        // Attempt to use SQL commands in sectionName fields
+        Course course = new Course("course-id", "name", "UTC", "institute");
+        coursesDb.createCourse(course);
+        String sectionName = "section'; DROP TABLE courses; --";
+        Section section = new Section(course, sectionName);
+
+        // The system should treat the input as a plain text string
+        coursesDb.createSection(section);
+
+        // Check that we are still able to get courses
+        Course actualCourse = coursesDb.getCourse("course-id");
+        assertEquals(course, actualCourse);
+    }
+
+    @Test
+    public void testSqlInjectionInGetSectionByName() throws Exception {
+        ______TS("SQL Injection test in getSectionByName");
+
+        Course course = new Course("course-id", "course-name", "UTC", "institute");
+        coursesDb.createCourse(course);
+        String sectionName = "section-name";
+        Section section = new Section(course, sectionName);
+
+        coursesDb.createSection(section);
+        Section actual = coursesDb.getSectionByName("course-id", "section'; DROP TABLE courses; --");
+        assertEquals(null, actual);
```

Comment:
The section name should match if we want the assertion to prove the name was sanitised for SQL. This applies for the team name too in `testSqlInjectionInGetSectionByCourseIdAndTeam` and `testSqlInjectionInGetTeamByName`.